app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if !logged_in?
+      redirect to '/login'
+    end
+    @tweets = Tweet.all
+    erb :'tweets/tweets'
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/create_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.create(params)
+    @tweet.user = current_user
+    if @tweet.save
+      redirect to "/tweets"
+    else
+      redirect to "/tweets/new"
+    end
+  end
+
+  get '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params[:id])
+    if logged_in?
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find_by_id(params[:id])
+    if logged_in?
+      erb :'/tweets/edit_tweet'
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?